File: src/main/scala/br/com/fabioaraujo/calcula/produto/dao/ProdutoRepository.scala
package br.com.fabioaraujo.calcula.produto.dao

import br.com.fabioaraujo.calcula.produto.dto.Produto
import org.springframework.data.repository.CrudRepository
import org.springframework.stereotype.Repository

@Repository
trait ProdutoRepository extends CrudRepository[Produto, java.lang.Long] {
  def findItemByNome(nome: String): Produto
}

File: src/main/java/br/com/fabioaraujo/calcula/produto/dao/ProdutoRepository.java
package br.com.fabioaraujo.calcula.produto.dao;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

import br.com.fabioaraujo.calcula.produto.dto.Produto;

@Repository
public interface ProdutoRepository extends CrudRepository<Produto, Long> {
  public Produto findItemByNome(String nome);
}